celavi/costgraph.py

```diff
@@ -797,14 +591,16 @@ def build_supplychain_graph(self):
             print(f'CostGraph: Calculating edge costs took {np.round((time() - _ctime)/60, 2)} minutes', flush=True)
 
         _cost_adjust = min([value for key, value in nx.get_edge_attributes(self.supply_chain, 'cost').items()])
-        self.cost_adjustment_factor[self.year] = abs(_cost_adjust) if _cost_adjust < 0.0 else 0.0
-
-        for edge in self.supply_chain.edges():
-            self.supply_chain.edges[edge]['cost'] = self.supply_chain.edges[edge]['cost'] + self.cost_adjustment_factor[self.year]
+        if _cost_adjust < 0.0:
```

Comment:
I see... This is where we need to adjust for negative costs because the shortest path algorithm does not work with negative values? I wonder if there are alternatives shortest path algorithms that work with negative value... Maybe there are also some other ways we could deal with negative values... Definitelly something to think of later on though...

Reply:
I reviewed all the path algorithms available in networkx and they all break with negative edge/weight/path values. I think because if there are some edges with negative weights, it's possible to have two distinct paths with identical weights and thus no "shortest" path. The cost adjustment is printed out during running and saved to a file - I'm not super happy with manually altering the cost values but the adjustment is reasonably transparent & can be un-done in postprocessing.